### PR TITLE
user_view_user_course

### PR DESCRIPTION
**Pull request by hienmvvn**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -35,17 +36,15 @@ gem "sdoc", "~> 0.4.0", group: :doc
 
 group :development, :test do
   gem "sqlite3"
+  gem "byebug"
+  gem 'fabrication'
```
Comment (ngocquyhoang) [inline Gemfile]: Using " "

File: app/assets/javascripts/customs.js [hunk 2]
```diff
@@ -12,3 +12,9 @@ $(document).ready(function(){
       event.preventDefault();
   });
 });
+
+$(document).ready(function(){
+    $('#change').click(function(){
```
Comment (ngocquyhoang) [inline app/assets/javascripts/customs.js]: Using " "

File: app/assets/javascripts/customs.js [hunk 3]
```diff
@@ -12,3 +12,9 @@ $(document).ready(function(){
...
+        $('#password').slideToggle('slow');
```
Comment (ngocquyhoang) [inline app/assets/javascripts/customs.js]: Using " "

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class UsersController < ApplicationController
+  def show
+	@courses = current_user.courses.all
+  end
+
```
Comment (ngocquyhoang) [inline app/controllers/users_controller.rb]: Delete this line

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  def course_status course
+    if course.free?
+      content_tag :div, "#{course.title}", class: "btn btn-default"
+    else course.in_process?
+      link_to("#") do
+        content_tag :div, "#{course.title}", class: "btn btn-success"
+      end
+    end
+  end
+
```
Comment (ngocquyhoang) [inline app/helpers/users_helper.rb]: delete this line

File: app/models/subject.rb [hunk 6]
```diff
@@ -11,11 +11,12 @@ class Subject < ActiveRecord::Base
 
   validates :title, presence: true, length: {minimum: 9, maximum: 90}
   validates :description, presence: true, length: {minimum: 9}
-  validate :validate
+  # validate :validate
```
Comment (ngocquyhoang) [inline app/models/subject.rb]: remove comment

File: db/schema.rb [hunk 7]
```diff
@@ -11,11 +11,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-<<<<<<< HEAD
```
Comment (ngocquyhoang) [inline db/schema.rb]: fix conflic

File: app/assets/javascripts/custom.js [hunk 8]
```diff
@@ -1,5 +0,0 @@
-$(document).ready(function(){
-    $('#change').click(function(){
-        $('#password').slideToggle('slow');
```
Comment (ngocquyhoang) [inline app/assets/javascripts/custom.js]: using number is bester for text

File: app/assets/javascripts/customs.js (hunk 2 above)
Comment (hienmvvn) [inline app/assets/javascripts/customs.js]: js dung ''

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+	@courses = current_user.courses.all
```
Comment (tanphan313) [inline app/controllers/users_controller.rb]: tab tab :v:

File: app/models/subject.rb (hunk 6 above)
Comment (hienmvvn) [inline app/models/subject.rb]: em để # validate :validate chạy seed anh ạ

File: app/assets/javascripts/customs.js (hunk 3 above)
Comment (ngocquyhoang) [inline app/assets/javascripts/customs.js]: using number is better than text , default is slow is 600ms

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+	@courses = current_user.courses
```
Comment (tanphan313) [inline app/controllers/users_controller.rb]: tab

Comment (ngocquyhoang) [inline app/controllers/users_controller.rb]: indent

File: app/helpers/users_helper.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+module UsersHelper
+  def course_status course
+    if course.free?
+      content_tag :div, "#{course.title}", class: "btn btn-default"
+    else course.in_process?
```
Comment (tanphan313) [inline app/helpers/users_helper.rb]: else hay elsif

File: app/assets/javascripts/customs.js (hunk 2 above)
Comment (ngocquyhoang) [inline app/assets/javascripts/customs.js]: Indent

File: app/models/course.rb [hunk 12]
```diff
@@ -1,4 +1,5 @@
 class Course < ActiveRecord::Base
+  enum status: [:free, :in_process, :closed]
   has_many :course_subjects
```
Comment (tanphan313) [inline app/models/course.rb]: phân tách ra chia group rõ ràng hơn

File: db/seeds.rb [hunk 13]
```diff
@@ -1,7 +1,20 @@
...
-#
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+4.times do
+  Fabricate :user, name: Faker::Name.name, email: Faker::Internet.free_email,
+    password: Faker::Internet.password(8), role: Faker::Number.between(0, 2)
+end
+
+10.times do
+  subject = Fabricate :subject, title: Faker::Lorem.characters(10), description: Faker::Lorem.sentence(3)
```
Comment (ngocquyhoang) [inline db/seeds.rb]: less than 80 character

File: app/models/course.rb [hunk 14]
```diff
@@ -1,4 +1,5 @@
 class Course < ActiveRecord::Base
+  enum status: [:free, :in_process, :closed]
```
Comment (ngocquyhoang) [inline app/models/course.rb]: block code

**Comment by ngocquyhoang** [comment]
Ok

**Comment by tanphan313** [comment]
OK

File: app/assets/javascripts/customs.js [hunk 15]
```diff
@@ -1,14 +1,20 @@
 $(document).ready(function(){
   $('form').on('click', '.remove_fields', function(event){
-      $(this).prev("input[type=hidden]").val(true);
-      $(this).parent().parent().parent().hide();
-      event.preventDefault();
+    $(this).prev("input[type=hidden]").val(true);
```
Comment (dieunb) [inline app/assets/javascripts/customs.js]: single quote

File: app/models/subject.rb (hunk 6 above)
Comment (dieunb) [inline app/models/subject.rb]: để tên khó hiểu vậy em?

Comment (hienmvvn) [inline app/models/subject.rb]: dạ vâng a

**Comment by tanphan313** [comment]
ok

**Pull request merged**